modules/openapi-generator/src/main/resources/powershell-experimental/http_signature_auth.mustache

```diff
@@ -0,0 +1,162 @@
+    }
+
+    foreach ($item in $ApiKeysList.GetEnumerator()) {
+        if (![string]::IsNullOrEmpty($item.Name)) {
+            if (Test-Path -Path $item.Value) {
+                $apiKey = $item.Value
+                $apikeyId = $item.Name
+                break;
+            }
+            else {
+                throw "API key file path does not exist."
+            }
+        }
+    }
+
+    if ($ApiKeyPrefixList.ContainsKey($apikeyId)) {
+        $apiPrefix = ApiKeyPrefixList[$apikeyId]
+    }
+
+    if ($apikeyId -and $apiKey -and $apiPrefix) {
+        $result = New-Object -Type PSCustomObject -Property @{ 
+            ApiKeyId       = $apikeyId;
+            ApiKeyFilePath = $apiKey
+            ApiKeyPrefix   = $apiPrefix
+        }
+    }
+    else {
+        return $null
+    }
+    return $result
+}
+
+<#
+.SYNOPSIS
+    Gets the headers for http signed auth.
+
+.DESCRIPTION
+    Gets the headers for the http signed auth. It use (targetpath), date, host and body digest to create authorization header.
+.PARAMETER Method
+    Http method
+.PARAMETER UriBuilder
+    UriBuilder for url and query parameter
+.PARAMETER Body
+    Request body
+.OUTPUTS
+    Hashtable
+#>
+function Get-{{{apiNamePrefix}}}HttpSignedHeader {
+    param(
+        [string]$Method,
+        [System.UriBuilder]$UriBuilder,
+        [string]$Body
+    )
+
+    #Hash table to store singed headers
+    $HttpSignedHeader = @{}
+    $TargetHost = $UriBuilder.Host
+
+    #Check for Authentication type
+    $apiKeyInfo = Get-{{{apiNamePrefix}}}APIKeyInfo
+    if ($null -eq $apiKeyInfo) {
+        throw "Unable to reterieve the api key info "
+    }
+
+    #get the body digest
+    $bodyHash = Get-{{{apiNamePrefix}}}StringHash -String $Body
+    $Digest = [String]::Format("SHA-256={0}", [Convert]::ToBase64String($bodyHash))
+
+    #get the date in UTC
+    $dateTime = Get-Date
+    $currentDate = $dateTime.ToUniversalTime().ToString("r")
+
+    $requestTargetPath = [string]::Format("{0} {1}{2}",$Method.ToLower(),$UriBuilder.Path.ToLower(),$UriBuilder.Query)
+    $h_requestTarget = [string]::Format("(request-target): {0}",$requestTargetPath)
+    $h_cdate = [string]::Format("date: {0}",$currentDate)
+    $h_digest = [string]::Format("digest: {0}",$Digest)
+    $h_targetHost = [string]::Format("host: {0}",$TargetHost)
+
+    $stringToSign = [String]::Format("{0}`n{1}`n{2}`n{3}",
+                                    $h_requestTarget,$h_cdate,
+                                    $h_targetHost,$h_digest)
+    
+    $hashedString = Get-{{{apiNamePrefix}}}StringHash -String $stringToSign
+    $signedHeader = Get-{{{apiNamePrefix}}}RSASHA256SignedString -APIKeyFilePath $apiKeyInfo.ApiKeyFilePath -DataToSign $hashedString
+    $authorizationHeader = [string]::Format("{0} keyId=""{1}"",algorithm=""rsa-sha256"",headers=""(request-target) date host digest"",signature=""{2}""", 
```

Comment:
This should not be hard-coded to "rsa-sha256" because 1) this algorithm has been deprecated in the HTTP signature spec and 2) the client should be able to decide which signature algorithm to use.